src/pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -175,6 +178,9 @@ def _set_ckpt_path(
                 return None
             ckpt_path = max(candidates_ts.keys(), key=partial(operator.getitem, candidates_ts))
 
+        if ckpt_path == "hpc":
```

Comment:
Should be an `elif`, no? Same for the checks above

Reply:
I don't think it really matters, since `ckpt_path` will be overwritten within those checks and I honestly find it a bit easier to follow. But, sure, let's make it a bunch of `elif`s, I don't necessarily hold strong opinion on that.